apm-agent-core/src/main/java/co/elastic/apm/agent/impl/CloudMetadataProvider.java

```diff
@@ -370,10 +372,12 @@ static CloudProviderInfo deserializeAzureMetadata(@Nullable String rawMetadata)
 
         CloudProviderInfo cloudProviderInfo = new CloudProviderInfo("azure");
         cloudProviderInfo.setAccount(new CloudProviderInfo.ProviderAccount(subscriptionId));
-        cloudProviderInfo.setInstance(new CloudProviderInfo.ProviderInstance(vmId, vmName));
-        cloudProviderInfo.setProject(new CloudProviderInfo.ProviderProject(resourceGroupName));
+        cloudProviderInfo.setInstance(new CloudProviderInfo.NameAndIdField(vmName, vmId));
```

Comment:
[minor] we could maybe make syntax a bit lighter here we by having `cloudProviderInfo.setInstance(String vmName, String vmId)`, so the setters would be mostly using Strings as parameters and the getters would return the built `NameAndIdField`.

Reply:
I started applying this suggestion, but even if it made the syntax "lighter", it made the API more complex, as for each of these you need to support both textual and numeric ID, which means duplicating setters...
So I left it as is.